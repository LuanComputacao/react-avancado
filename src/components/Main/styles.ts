'use client'

import styled from 'styled-components'

export const Wraper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`
