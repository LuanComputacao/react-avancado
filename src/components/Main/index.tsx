import React from 'react'
import * as S from './styles'

const Main = () => (
  <S.Wraper>
    <h1>LuanComputacao</h1>
  </S.Wraper>
)

export default Main
