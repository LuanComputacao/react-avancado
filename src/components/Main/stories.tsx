import { Meta, StoryObj } from '@storybook/react'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  parameters: {
    layout: 'fullscreen'
  },
  args: {
    name: 'Luan Roger'
  }
} as Meta

export const Default: StoryObj = {}

export const Secondary: StoryObj = {
  args: {
    name: 'Luan Roger Santos Santana'
  }
}
