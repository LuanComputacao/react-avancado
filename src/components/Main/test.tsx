import React from 'react' // <-- Add this line
import { render, screen } from '@testing-library/react'
import Main from './index'

describe('<Main />', () => {
  it('should render the heading', () => {
    // render the component
    const { container } = render(<Main name="LuanComputacao" />)
    // find the heading element
    expect(
      screen.getByRole('heading', { name: /luancomputacao/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})

it('should render the colors correctly', () => {
  // render the component
  const { container } = render(<Main name="Some name" />)
  // find the heading element
  expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
})
