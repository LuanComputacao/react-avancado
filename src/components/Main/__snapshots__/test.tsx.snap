// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

<main
  class="c0"
>
  <h1>
    LuanComputacao
  </h1>
</main>
`;
